### ISSUE TITLE
Unable to concretize with Clingo when libyogrt is part of dependency tree

### ISSUE THREAD

**Issue by joshessman-llnl** [LLNL/spack#20611]
### Steps to reproduce the issue
Any of the above result in the same error:
```console
$ spack spec -I libyogrt
$ spack spec -I scr # SCR depends on libyogrt
$ spack spec -I axom # axom depends on SCR
$ spack spec -I macsio # macsio depends on SCR
...
```

### Error Message

```console
Concretized
--------------------------------
==> Error: invalid values for variant "scheduler" in package "libyogrt": ['lsf']
```

I imagine this is because https://github.com/spack/spack/blob/c22141f444861abeaee297a3d92696e9ae94a509/var/spack/repos/builtin/packages/libyogrt/package.py#L39

references an invalid value of the 'scheduler` variant:
https://github.com/spack/spack/blob/c22141f444861abeaee297a3d92696e9ae94a509/var/spack/repos/builtin/packages/libyogrt/package.py#L36

Adding `lsf` to the possible values for `scheduler` fixes the issue, but I am not sure that this fix is correct.

### Information on your system

* **Spack:** 0.16.0
* **Python:** 3.7.2
* **Platform:** linux-rhel7-power9le
* **Concretizer:** clingo

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
This issue is correctly diagnosed in its description. The new concretizer is stricter in certain checks if compared to the original concretizer. Here it fails since the `scheduler=lsf` anonymous spec used in a `conflicts` directive is not an allowed state of `libyogrt`.

**Comment by alalazo** [comment]
This should be solved by #20794

**Comment by joshessman-llnl** [comment]
Awesome, thank you @alalazo!
